slapo/model_schedule/gpt2.py

```diff
@@ -478,6 +478,10 @@ def shard_parameters(
                 sub_sch["module.out_proj"].sync(
                     mode="fwd_post", sync_op_or_fn="reduce_scatter", axis=1
                 )
+
+                # In this case, both attention droput and residual dropout
+                # have to use different random seeds.
+                sub_sch["module"].fork_rng()
```

Comment:
with sequence parallel, we do not need to specify fork rng since different TP ranks already use different seeds

Reply:
We have 2 options:
1. [Megatron] Apply .fork_rng() to top modules (embedding, transformers, pooler). The RNG by default uses the same seed within a TP group.
2. [Ours] The RNG has a mode to use different seeds within a TP group. With that, we don't need any .fork_rng() when sequence parallelism is enabled.

I was trying to use (1), but it turns out that it's tricky in Slapo, because `.fork_rng()` is s context manager, which cannot be traced. In other words, if we apply `sch.fork_rng()`, then we cannot trace the top module for pipeline partitioning. However, if we want to trace the top module and then apply `.fork_rng()`, we have to change the implementation in pipeline partitioning, because the current implementation of pipeline partitioning assume top modules are traced. Considering the efforts, I decided to go with (2) for now.